Standards/scs-0217-v1-baseline-cluster-security.md

```diff
@@ -0,0 +1,141 @@
+---
+title: Kubernetes cluster baseline security
+type: Standard
+status: Draft
+track: KaaS
+---
+
+## Introduction
+
+A Kubernetes instance is provided as a cluster, which consists of a set of machines,
+so-called nodes. These clusters are configured depending on the needs of its users, which
+is possible due to the flexibility of Kubernetes.
+Due to the regular changes, there are always new security features to deploy and use in Kubernetes.
+Nevertheless, a provider (or even a customer) needs to take action in order to achieve a
+hardened, secure cluster due to the myriad of configurations possible. This is especially
+the case since Kubernetes ships with insecure features and configurations out of the box,
+which will need to be mitigated by an administrator with the proper knowledge.
+Hardened, secure Kubernetes clusters are desirable regardless of the possible threat model,
+since higher security doesn't necessarily mean higher complexity in this case.
+
+## Motivation
+
+Kubernetes clusters are highly configurable, which also gives rise to different security
+problems, if the configuration isn't done properly.
+These security risks can potentially be exposed in many different parts of a cluster, e.g.
+different APIs, authorization and authentication procedures or even Pod privilege mechanisms.
+In order to mitigate these problems, different steps and mechanisms could be used to increase
+the security of a Kubernetes setup.
+
+## Design Considerations
+
+### External CA
+
+Kubernetes provides an API to provision TLS certificates that can be signed by a CA.
+This CA can be controlled by the cluster provider, which enables much more tight control
+over the clusters communication and therefore also better controllable security.
+
+In order to do this, the CA certificate bundle needs to be added to the trusted certificates
+of the server.
+To provide a certificate, the following steps need to be undertaken:
+
+1. Create a CSR
```

Comment:
abrivation "CSR" is not introduced.

```suggestion+
1. Create a CSR (Certificate Signing Request)
```

Reply:
Added a `Terminology` section in https://github.com/SovereignCloudStack/standards/pull/376/commits/5610fe15c9cf2ad9ddf1ddf89f0d575a9002b5d2